tests/test_single_topology_combined.py

```diff
@@ -179,3 +179,42 @@ def _test_combined_parameters_impl_nonbonded(host_system_omm):
                 assert ref_q == test_q
                 assert test_sig == ref_sig
                 assert test_eps == ref_eps
+
+
+@pytest.mark.parametrize("host_system_fixture", ["solvent_host_system", "complex_host_system"])
+def test_combined_parameters_nonbonded_intermediate(
+    host_system_fixture, hif2a_ligand_pair_single_topology: SingleTopology, request
+):
+    st = hif2a_ligand_pair_single_topology
+    host_sys, host_masses = request.getfixturevalue(host_system_fixture)
+    num_host_atoms = len(host_masses)
+
+    rng = np.random.default_rng(2022)
+
+    for lamb in rng.uniform(0.01, 0.99, (10,)):
+        hgs = st.combine_with_host(host_sys, lamb=lamb)
+        cutoff = hgs.nonbonded_host_guest.get_cutoff()
+
+        assert hgs.nonbonded_host_guest.params is not None
+        guest_params = hgs.nonbonded_host_guest.params[num_host_atoms:]
+        ws_core = [w for flag, (_, _, _, w) in zip(st.c_flags, guest_params) if flag == 0]
+        ws_a = [w for flag, (_, _, _, w) in zip(st.c_flags, guest_params) if flag == 1]
+        ws_b = [w for flag, (_, _, _, w) in zip(st.c_flags, guest_params) if flag == 2]
+
+        # core atoms fixed at w = 0
+        assert all(w == 0.0 for w in ws_core)
+
+        # dummy groups have consistent w coords
+        assert len(set(ws_a)) == 1
+        assert len(set(ws_b)) == 1
+        (w_a,) = set(ws_a)
+        (w_b,) = set(ws_b)
+
+        # w in [0, cutoff]
+        assert 0 < w_a < cutoff
+        assert 0 < w_b < cutoff
+
+        if lamb < 0.5:
```

Comment:
are we still setting `w_b` to negative values here? (also, this may change if we do double insertion, i.e. at `lamb=0.5` `w_a=w_b=0` in certain cases)

Reply:
> are we still setting w_b to negative values here?

No, `w_b` is now positive (+cutoff -> 0) as of this PR (see [here](https://github.com/proteneer/timemachine/pull/900/files#diff-0b9bd14a098c7a129a18f14ffb94d84fc8e5c1020d1a55d0e6f53d545c1a4084R1218)).

Reply:
> also, this may change if we do double insertion, i.e. at lamb=0.5 w_a=w_b=0 in certain cases

To clarify, this would pertain to a future version of the SingleTopology class, right? (I don't think we currently have this capability)

Reply:
correct - just something to remember later on when we break this test!